crates/sui-genesis-builder/src/lib.rs

```diff
@@ -914,7 +914,8 @@ fn create_genesis_transaction(
         let expensive_checks = false;
         let certificate_deny_set = HashSet::new();
         let transaction_data = &genesis_transaction.data().intent_message().value;
-        let (kind, signer, _) = transaction_data.execution_parts();
+        let (kind, signer, mut gas_data) = transaction_data.execution_parts();
+        gas_data.payment = vec![];
```

Comment:
Might also be worth setting `sponsor` to None as well here.

Reply:
if I understand the comment (please correct me if I am wrong) sponsor is not in `GasData`.
`GasData` carries the `owner` which could be the sender or the sponsor.
So we leave it untouched here as I think should be?